src/includes/txs/base_tx.h

```diff
@@ -49,9 +48,23 @@ int size_tx(Transaction *tx);
  */
 void hash_tx(unsigned char *dest, Transaction *tx);
 
+/**
+ * @brief Free transaction
+ *
+ * @param tx Transaction to free
+ */
+void free_tx(Transaction *tx);
+
+/**
+ * @brief Copy transaction
+ *
+ * @param tx Transaction to copy
+ * @return Copy of passed transcation
+ */
+Transaction *copy_tx(Transaction *tx);
```

Comment:
maybe this should do a check if transaction is NULL and take a pointer to a new transaction ? and then return an int if the transaction can be copied?

If not, add a note that tX should be null checked prior

Reply:
Will change to return NULL if passed NULL